internal/services/apimanagement/migration/api_v0_to_v1.go

```diff
@@ -0,0 +1,329 @@
+			},
+		},
+
+		"subscription_required": {
+			Type:     pluginsdk.TypeBool,
+			Optional: true,
+			Default:  true,
+		},
+
+		"terms_of_service_url": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ValidateFunc: validation.IsURLWithHTTPorHTTPS,
+		},
+
+		"source_api_id": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ValidateFunc: validate.ApiID,
+		},
+
+		"oauth2_authorization": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			MaxItems: 1,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"authorization_server_name": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: validate.ApiManagementChildName,
+					},
+					"scope": {
+						Type:     pluginsdk.TypeString,
+						Optional: true,
+						// There is currently no validation, as any length and characters can be used in the field
+					},
+				},
+			},
+		},
+
+		"openid_authentication": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			MaxItems: 1,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"openid_provider_name": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: validate.ApiManagementChildName,
+					},
+					"bearer_token_sending_methods": {
+						Type:     pluginsdk.TypeSet,
+						Optional: true,
+						Elem: &pluginsdk.Schema{
+							Type: pluginsdk.TypeString,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(apimanagement.BearerTokenSendingMethodsAuthorizationHeader),
+								string(apimanagement.BearerTokenSendingMethodsQuery),
+							}, false),
+						},
+					},
+				},
+			},
+		},
+
+		// Computed
+		"is_current": {
+			Type:     pluginsdk.TypeBool,
+			Computed: true,
+		},
+
+		"is_online": {
+			Type:     pluginsdk.TypeBool,
+			Computed: true,
+		},
+
+		"version": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+			Optional: true,
+		},
+
+		"version_description": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"version_set_id": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+			Optional: true,
+		},
+	}
+
+	if !features.FourPointOhBeta() {
+		schema["api_type"].ConflictsWith = []string{"soap_pass_through"}
+
+		schema["soap_pass_through"] = &pluginsdk.Schema{
+			Type:          pluginsdk.TypeBool,
+			Optional:      true,
+			Computed:      true,
+			Deprecated:    "`soap_pass_through` will be removed in favour of the property `api_type` in version 4.0 of the AzureRM Provider",
+			ConflictsWith: []string{"api_type"},
+		}
+	}
+
+	return schema
+}
+
+func (ApiV0ToV1) UpgradeFunc() pluginsdk.StateUpgraderFunc {
+	return func(ctx context.Context, rawState map[string]interface{}, meta interface{}) (map[string]interface{}, error) {
+		rawState["id"] = fmt.Sprintf("%s;rev=%s", rawState["id"].(string), rawState["revision"].(string))
```

Comment:
I'm not sure this transformation is sufficient I think this may need to be 
```suggestion
		apiId := fmt.Sprintf("%s;rev=%s", rawState["name"], revision)
		oldId, err := parse.ApiID(rawState["id"].(string))
		if err != nil {
			return rawState, err
		}
		newId := parse.NewApiID(oldId.SubscriptionId, oldId.ResourceGroupName, oldId.ServiceName, apiId).ID()
		log.Printf("[DEBUG] Updating ID from %q to %q", oldId, newId)
		rawState["id"] = newId
```

Reply:
Updated to match your implementation